pkg/volume/util/operationexecutor/operation_generator_test.go

```diff
@@ -89,57 +90,46 @@ func TestOperationGenerator_GenerateUnmapVolumeFunc_PluginName(t *testing.T) {
 			"volume_plugin":  expectedPluginName,
 		}
 		// compare the relative change of the metric because of the global state of the prometheus.DefaultGatherer.Gather()
-		storageOperationStatusCountMetricBefore := findMetricWithNameAndLabels(metricFamilyName, labelFilter)
+		storageOperationStatusCountMetricBeforeValue, beforeErr := getMetricCounterValueWithNameAndLabels(metricFamilyName, labelFilter)
 
 		var ee error
 		unmapVolumeFunc.CompleteFunc(&ee)
 
-		storageOperationStatusCountMetricAfter := findMetricWithNameAndLabels(metricFamilyName, labelFilter)
-		if storageOperationStatusCountMetricAfter == nil {
-			t.Fatalf("Couldn't find the metric with name(%s) and labels(%v)", metricFamilyName, labelFilter)
+		storageOperationStatusCountMetricAfterValue, afterErr := getMetricCounterValueWithNameAndLabels(metricFamilyName, labelFilter)
+		if afterErr != nil {
+			t.Fatalf("%s", afterErr)
 		}
 
-		if storageOperationStatusCountMetricBefore == nil {
-			assert.Equal(t, float64(1), *storageOperationStatusCountMetricAfter.Counter.Value, tc.name)
+		if beforeErr != nil {
+			assert.Equal(t, float64(1), storageOperationStatusCountMetricAfterValue, tc.name)
 		} else {
-			metricValueDiff := *storageOperationStatusCountMetricAfter.Counter.Value - *storageOperationStatusCountMetricBefore.Counter.Value
+			metricValueDiff := storageOperationStatusCountMetricAfterValue - storageOperationStatusCountMetricBeforeValue
 			assert.Equal(t, float64(1), metricValueDiff, tc.name)
 		}
 	}
 }
 
-func findMetricWithNameAndLabels(metricFamilyName string, labelFilter map[string]string) *io_prometheus_client.Metric {
-	metricFamily := getMetricFamily(metricFamilyName)
+// getMetricCounterValueWithNameAndLabels returns metric value according to metricFamilyName and labelFilter
+func getMetricCounterValueWithNameAndLabels(metricFamilyName string, labelFilter map[string]string) (float64, error) {
```

Comment:
/cc @logicalhan 
how about this ?  We just need to extract the common function GetMetricFamily() in metric .